tests/e2e/runner/__tests__/minter-suite-v2/i-shared-minter-v0.test.ts

```diff
@@ -0,0 +1,186 @@
+  );
+const linkLibraryAddresses: GenArt721CoreV3LibraryAddresses = {
+  "contracts/libs/0.8.x/BytecodeStorageV1.sol:BytecodeStorageReader":
+    bytecodeStorageReaderAddress,
+};
+const genArt721CoreContract = new GenArt721CoreV3__factory(
+  linkLibraryAddresses,
+  deployer
+).attach(genArt721CoreAddress);
+
+// get MinterSetPriceV5
+if (!config.iSharedMinterV0Contracts) {
+  throw new Error("No iSharedMinterV0Contracts in config");
+}
+const minterSetPriceV5Address = config.iSharedMinterV0Contracts[0].address;
+const minterSetPriceV5Contract = new MinterSetPriceV5__factory(deployer).attach(
+  minterSetPriceV5Address
+);
+
+describe("iSharedMinterV0 event handling", () => {
+  beforeAll(async () => {
+    await waitUntilSubgraphIsSynced(client);
+  });
+
+  describe("Indexed after setup", () => {
+    test("created new Minter during deployment and allowlisting", async () => {
+      const targetId = minterSetPriceV5Address.toLowerCase();
+      const minterRes = await getMinterDetails(client, targetId);
+      expect(minterRes.id).toBe(targetId);
+    });
+  });
+
+  describe("PricePerTokenInWeiUpdated", () => {
+    afterEach(async () => {
+      // clear the minter for project zero
+      try {
+        await sharedMinterFilterContract
+          .connect(artist)
+          .removeMinterForProject(0, genArt721CoreAddress);
+      } catch (error) {
+        // swallow error in case of test failure
+      }
+    });
+
+    test("Price is updated and configured", async () => {
+      // set minter for project zero to the target minter
+      await sharedMinterFilterContract
+        .connect(artist)
+        .setMinterForProject(0, genArt721CoreAddress, minterSetPriceV5Address);
+      // update price
+      const newPrice = ethers.utils.parseEther(Math.random().toString());
+      await minterSetPriceV5Contract
+        .connect(artist)
+        .updatePricePerTokenInWei(0, genArt721CoreAddress, newPrice);
+      await waitUntilSubgraphIsSynced(client);
+      // validate price in subgraph
+      const targetId = `${minterSetPriceV5Address.toLowerCase()}-${genArt721CoreAddress.toLowerCase()}-0`;
+      const minterConfigRes = await getProjectMinterConfigurationDetails(
+        client,
+        targetId
+      );
+      expect(minterConfigRes.basePrice).toBe(newPrice.toString());
+      // @dev this could have been true before the update, but we can't know for sure, and
+      // at least we validate that it's true after the update
+      expect(minterConfigRes.priceIsConfigured).toBe(true);
+    });
+  });
+
+  // TODO - need ERC20 minter to test event handling of "ProjectCurrencyInfoUpdated"
+  // ERC20 minter is not yet developed at the time of writing this test
+  // describe("ProjectCurrencyInfoUpdated", () => {
+  //   afterEach(async () => {
+  //     // TODO
+  //   });
+
+  //   test("Currency is updated and configured", async () => {
+  //     // TODO - need ERC20 minter to test this
+  //   });
+  // });
```

Comment:
Can we please add an associated Asana ticket linked here for this TODO?

Reply:
yeah it is added in #235 (which is tracked in Asana here: https://app.asana.com/0/0/1205078004803554/f)
- @jakerockland would it be best to commit the asana link to our codebase in these situations?

Reply:
oh, I see what you were saying - yep we have it in asana (and already in a PR up for review). resolving 😄